packages/bridge/src/axelar/__tests__/axelar-bridge-provider.spec.ts

```diff
@@ -0,0 +1,465 @@
+      toChain: { chainId: "43114", chainName: "Avalanche", chainType: "evm" },
+      transferFee: {
+        amount: "0.01",
+        denom: "ETH",
+        sourceDenom: "eth",
+        decimals: 18,
+      },
+      estimatedGasFee: {
+        amount: "420000000000000",
+        denom: "ETH",
+        sourceDenom: "ETH",
+        decimals: 18,
+      },
+    });
+  });
+
+  it("should throw an error if the amount surpasses the transfer limit", async () => {
+    const mockDepositClient: Partial<AxelarAssetTransfer> = {
+      getDepositAddress: jest.fn().mockResolvedValue("0x123"),
+    };
+    const mockQueryClient: Partial<AxelarQueryAPI> = {
+      getTransferFee: jest.fn().mockResolvedValue({
+        fee: {
+          amount: "0.01",
+          denom: "ETH",
+        },
+      }),
+      getTransferLimit: jest.fn().mockResolvedValue("1"),
+    };
+
+    jest
+      .spyOn(provider, "getAssetTransferClient")
+      .mockResolvedValue(mockDepositClient as AxelarAssetTransfer);
+    jest
+      .spyOn(provider, "getQueryClient")
+      .mockResolvedValue(mockQueryClient as AxelarQueryAPI);
+
+    await expect(
+      provider.getQuote({
+        fromChain: { chainId: "1", chainName: "Ethereum", chainType: "evm" },
+        toChain: { chainId: "43114", chainName: "Avalanche", chainType: "evm" },
+        fromAsset: {
+          denom: "ETH",
+          address: "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
+          decimals: 18,
+          sourceDenom: "eth",
+        },
+        toAsset: {
+          denom: "AVAX",
+          address: "0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7",
+          decimals: 18,
+          sourceDenom: "avax",
+        },
+        fromAmount: "1.1",
+        fromAddress: "0x742d35Cc6634C0532925a3b844Bc454e4438f44e",
+        toAddress: "0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7",
+        slippage: 1,
+      })
+    ).rejects.toThrow(
+      "Amount exceeds transfer limit of 0.000000000000000001 ETH"
+    );
+  });
+
+  it("should handle errors in getQuote", async () => {
+    jest.spyOn(provider, "getQueryClient").mockImplementationOnce(() => {
+      throw new Error("Query client error");
+    });
+
+    await expect(
+      provider.getQuote({
+        fromChain: { chainId: "1", chainName: "Ethereum", chainType: "evm" },
+        toChain: { chainId: "43114", chainName: "Avalanche", chainType: "evm" },
+        fromAsset: {
+          denom: "ETH",
+          address: "0x0",
+          decimals: 18,
+          sourceDenom: "eth",
+        },
+        toAsset: {
+          denom: "AVAX",
+          address: "0x0",
+          decimals: 18,
+          sourceDenom: "avax",
+        },
+        fromAmount: "1",
+        fromAddress: "0x123",
+        toAddress: "0x456",
+        slippage: 1,
+      })
+    ).rejects.toThrow("Query client error");
+  });
+
+  it("should throw an error when withdrawing native asset without using 'autoUnwrapIntoNative'", async () => {
```

Comment:
I'm not sure if this falls within the provider's responsibility. I believe it will depend on the user's selection in the UI. If I remember correctly, with our revamp, ETH will be the target instead